services/ui-src/src/components/forms/AdminBannerForm.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { render, screen } from "@testing-library/react";
+// components
+import { AdminBannerForm } from "components";
+// utils
+import { RouterWrappedComponent } from "utils/testing/mockRouter";
+import userEvent from "@testing-library/user-event";
+import { testA11y } from "utils/testing/commonTests";
+
+const mockWriteAdminBanner = jest.fn();
+window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+const adminBannerFormComponent = (writeAdminBanner: Function) => (
+  <RouterWrappedComponent>
+    <AdminBannerForm
+      writeAdminBanner={writeAdminBanner}
+      data-testid="test-form"
+    />
+  </RouterWrappedComponent>
+);
+
+describe("<AdminBannerForm />", () => {
+  test("AdminBannerForm is visible", () => {
+    render(adminBannerFormComponent(mockWriteAdminBanner));
+    const form = screen.getByTestId("test-form");
+    expect(form).toBeVisible();
+  });
```

Comment:
I prefer to avoid `getByTestId` whenever possible, usually in favor of `getByRole`. In this case, there is no ARIA "form" role, so an equivalent test would have to check the various elements within the form, to ensure they are all visible. Something like
```ts
    expect(screen.getByRole("textbox", { name: "Title text"} })).toBeVisible();
    expect(screen.getByRole("textbox", { name: "Description text"} })).toBeVisible();
    expect(screen.getByRole("textbox", { name: "Link"} })).toBeVisible();
    expect(screen.getByRole("textbox", { name: "Start date"} })).toBeVisible();
    expect(screen.getByRole("textbox", { name: "End date"} })).toBeVisible();
    expect(screen.getByRole("button", { name: "Replace Current Banner"} })).toBeVisible();
```
This aligns more closely to how a user would interact with the page; they can't easily tell if these inputs are in a `<form>`, they can only tell whether the inputs are present on the page.

You might object that this becomes a really verbose & redundant test, and I would agree. For one, we don't have any conditional logic in `<AdminBannerForm>` for whether or not to display the form, so there isn't really a way for this test to fail. For two, the next test verifies all the inputs are present anyway, by typing in / clicking on them. In fact I _think_ `userEvent` also verifies inputs' _visibility_ before typing in them? So I would also be fine if we just delete this test.

Reply:
i went with making sure the elements is on the page since you wrote it out for me anyway